### PR TITLE
python310.pkgs.ucsmsdk: init at 0.9.14

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ucsmsdk/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "CiscoUcs";
+    repo = "ucsmsdk";
+    rev = "v${version}";
+    sha256 = "sha256-lSkURvKRgW+qV1A8OT4WYsMGlxxIqaFnxQ3Rnlixdw0=";
+  };
+
+  propagatedBuildInputs = [
+    pyparsing
+    setuptools
```
Comment (tjni) [inline pkgs/development/python-modules/ucsmsdk/default.nix:23]: Even though it's listed in `requirements.txt`, I don't think this is a necessary propagated dependency. Maybe we can remove it using `pythonRelaxDepsHook`.

File: pkgs/development/python-modules/ucsmsdk/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+buildPythonPackage rec {
+  pname = "ucsmsdk";
+  version = "0.9.14";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "CiscoUcs";
+    repo = "ucsmsdk";
+    rev = "v${version}";
+    sha256 = "sha256-lSkURvKRgW+qV1A8OT4WYsMGlxxIqaFnxQ3Rnlixdw0=";
```
Comment (tjni) [inline pkgs/development/python-modules/ucsmsdk/default.nix:18]: Nit: `sha256` -> `hash`

**Review by tjni — APPROVED** [review]
LGTM modulo two small comments

File: pkgs/development/python-modules/ucsmsdk/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/ucsmsdk/default.nix]: It isn't even reading that file and it just works without it.

File: pkgs/development/python-modules/ucsmsdk/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pyparsing
+, setuptools
```
Comment (tjni) [inline pkgs/development/python-modules/ucsmsdk/default.nix:5]: We can nix this too :)

**Review by tjni — APPROVED** [review]
Thank you!

**Review by tjni — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
thanks for helping out @tjni

**Pull request merged**